document-store/src/main/java/org/hypertrace/core/documentstore/mongo/query/parser/MongoRelationalExpressionParser.java

```diff
@@ -91,6 +99,16 @@ private static Map<String, Object> generateMap(
     return (lhs, rhs) -> {
       final String parsedLhs = lhs.accept(identifierParser);
       final Object parsedRhs = rhs.accept(rhsParser);
+      Map<String, String> parsedObject = Map.of();
+      if (parsedRhs instanceof String) {
+        final ObjectReader objectReader = new ObjectMapper().readerFor(Map.class);
```

Comment:
This logic can be as simple as we have done in `org.hypertrace.core.documentstore.mongo.subdoc.MongoSubDocumentValueParser.parse`
That comes with necessary sanitization, etc.